File: 447/447.go
package main

import "fmt"

func numberOfBoomerangs(points [][]int) int {
	ans := 0
	for _, p := range points {
		mp := make(map[int]int)
		for _, q := range points {
			dis := (p[0]-q[0])*(p[0]-q[0]) + (p[1]-q[1])*(p[1]-q[1])
			mp[dis]++
		}
		for _, v := range mp {
			ans += v * (v - 1)
		}
	}
	return ans
}

func main() {
	fmt.Println(numberOfBoomerangs([][]int{{1, 1}, {2, 2}, {3, 3}}))
}

File: 447/447.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int numberOfBoomerangs(vector<vector<int>>& points) {
        int ans = 0;
        for(int i=0;i<points.size();i++) {
            map<int,int> mp;
            map<int,int> ::iterator it;
            for(int j=0;j<points.size();j++) {
                int dis = (points[i][0] - points[j][0]) * (points[i][0] - points[j][0])  + (points[i][1] - points[j][1]) * (points[i][1] - points[j][1]);
                mp[dis]++;
            }
            for(it=mp.begin();it!=mp.end();it++) {
                ans += it->second * (it->second - 1);
            }
        }
        return ans;
    }
};